src/Mobs/Monster.cpp

```diff
@@ -133,16 +139,30 @@ bool cMonster::TickPathFinding(cChunk & a_Chunk)
 		return false;
 	}
 
+	double Deviation = (m_FinalDestination - m_PathFinderDestination).Length();  // Distance between where we're going and where we should go.
+	if (Deviation > 0.25)
+	{
+		double DistanceToTarget = (m_FinalDestination - GetPosition()).Length();
+		if (
+			((GetPosition() - m_PathFinderDestination).Length() < 0.25) ||  // If we reached the last path waypoint.
+			m_TicksSinceLastPathReset > 0.15 * DistanceToTarget * DistanceToTarget  // Interval is proportional to distance squared.
```

Comment:
Perhaps you could replace `DistanceToTarget * DistanceToTarget` with `(m_FinalDestination - GetPosition()).SqrLength()' and add parentheses around the comparison?